interceptors/logging/logging.go

```diff
@@ -177,14 +180,25 @@ func ExtractFields(ctx context.Context) Fields {
 //
 // Don't overuse mutation of fields to avoid surprises.
 func InjectFields(ctx context.Context, f Fields) context.Context {
-	return context.WithValue(ctx, fieldsCtxMarkerKey, f.WithUnique(ExtractFields(ctx)))
+	return context.WithValue(ctx, fieldsCtxMarkerKey, &fieldsCtxValue{f.WithUnique(ExtractFields(ctx))})
```

Comment:
Mind using explicit field names?
```suggestion
	return context.WithValue(ctx, fieldsCtxMarkerKey, &fieldsCtxValue{fields: f.WithUnique(ExtractFields(ctx))})
```